### PR TITLE
[build] Allow buildspec override from config

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Allow the buildspec file used for image builds in PR to be overriden using a config file

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [x] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [x] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -18,39 +18,7 @@ phases:
...
-            # to create a correct path to buildspec files
-            PATH_TO_BULDSPEC=$(echo "$FRAMEWORK" | tr _ /)
-            python src/main.py --buildspec $PATH_TO_BULDSPEC/buildspec.yml --framework $FRAMEWORK
-          elif expr "${CODEBUILD_BUILD_ARN}" : ".*habana.*" >/dev/null; then
-            python src/main.py --buildspec habana/$FRAMEWORK/buildspec.yml --framework $FRAMEWORK
-          else
-            python src/main.py --buildspec $FRAMEWORK/buildspec.yml --framework $FRAMEWORK
-          fi
-        fi
+      - python src/main.py --buildspec $FRAMEWORK_BUILDSPEC_FILE --framework $FRAMEWORK
```
Comment (arjkesh) [inline buildspec.yml:21]: FRAMEWORK_BUILDSPEC_FILE is a path I assume?

Comment (saimidu) [reply buildspec.yml:21]: Yeah, it is the file path.

File: src/codebuild_environment.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def get_github_repo_url():
+    """
+    Get env variable CODEBUILD_SOURCE_REPO_URL
+
+    :return: value or empty string if not set
+    """
+    # Example: "https://github.com/aws/deep-learning-containers.git"
+    return os.getenv("CODEBUILD_SOURCE_REPO_URL")
```
Comment (arjkesh) [inline src/codebuild_environment.py:25]: why empty string default for CODEBUILD_BUILD_ARN but not for this one?

File: dlc_build_version_override_config.toml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Assign the path to the required buildspec file from the deep-learning-containers folder
+
+# Standard Framework DLC buildspecs
```
[the remark this answers is not in the document]
Comment (saimidu) [reply dlc_build_version_override_config.toml:3]: This file is already long, and will get longer in the future. Therefore, in the interest of preventing confusion in changing the developer config file, I have separated the configurations to override the buildspec file into another toml file.

File: src/codebuild_environment.py (hunk 2 above)
Comment (saimidu) [reply src/codebuild_environment.py:25]: I copy pasted the getenv CODEBUILD_BUILD_ARN from
https://github.com/aws/deep-learning-containers/blob/16afa88d995b6f6afcaca3c80cba3d251e72eace/src/utils.py#L40

However, I copy-pasted the getenv for CODEBUILD_SOURCE_REPO_URL from
https://github.com/aws/deep-learning-containers/blob/16afa88d995b6f6afcaca3c80cba3d251e72eace/src/send_status.py#L77
and
https://github.com/aws/deep-learning-containers/blob/16afa88d995b6f6afcaca3c80cba3d251e72eace/src/utils.py#L130

File: dlc_build_version_override_config.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (arjkesh) [reply dlc_build_version_override_config.toml:7]: I actually think separate files is more confusing, this one also has no toml headers. Basically user will need to go through multiple files and edit them. Can you start by having one file and adding toml headers?

**Review by arjkesh — APPROVED** [review]

**Pull request merged**